x/protocolpool/keeper/keeper.go

```diff
@@ -90,6 +101,141 @@ func (k Keeper) GetCommunityPool(ctx context.Context) (sdk.Coins, error) {
 	return k.bankKeeper.GetAllBalances(ctx, moduleAccount.GetAddress()), nil
 }
 
+func (k Keeper) withdrawContinuousFund(ctx context.Context, recipient sdk.AccAddress) (sdk.Coin, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	cf, err := k.ContinuousFund.Get(ctx, recipient)
+	if err != nil {
+		if errors.Is(err, collections.ErrNotFound) {
+			return sdk.Coin{}, fmt.Errorf("no continuous fund found for recipient: %s", recipient.String())
+		}
+	}
+	if cf.Expiry.Before(sdkCtx.HeaderInfo().Time) {
```

Comment:
below we dont check if this is 0, do we assume 0 is unlimited?

Reply:
yes